source/core/utils/dns-ip-version.ts

```diff
@@ -0,0 +1,20 @@
+export type DnsLookupIpVersion = 'auto' | 'ipv4' | 'ipv6';
+type DnsIpFamily = 0 | 4 | 6;
+
+const conversionTable = {
+	auto: 0,
+	ipv4: 4,
+	ipv6: 6
+};
+
+export const isDnsLookupIpVersion = (value: any): boolean => {
+	return value in conversionTable;
+};
+
+export const dnsLookupIpVersionToFamily = (dnsLookupIpVersion: DnsLookupIpVersion): DnsIpFamily => {
+	if (isDnsLookupIpVersion(dnsLookupIpVersion)) {
+		return conversionTable[dnsLookupIpVersion] as DnsIpFamily;
+	}
+
+	throw new Error('Invalid DnsLookupIpVersion');
```

Comment:
```suggestion
	throw new Error('Invalid `dnsLookupIpVersion` option value`');
```

Reply:
That function should be generic, not related to `dnsLookupIpVersion`.
I think it's better to catch the error and throw "Invalid `dnsLookupIpVersion` option value" where `dnsLookupIpVersionToFamily` is used:

```
requestOptions.family = dnsLookupIpVersionToFamily(options.dnsLookupIpVersion);
```
```
try {
  requestOptions.family = dnsLookupIpVersionToFamily(options.dnsLookupIpVersion);
} catch {
  throw new Error('Invalid `dnsLookupIpVersion` option value');
}
```

Reply:
👍